### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by jpdeleon**
@enzoampil I added cache of all PSE companies from Jan 2010 to now that lives in /data/merged_stock_data.zip (size~4MB) based on new added function `fastquant.update_pse_data_cache` which in turn uses `get_pse_data_old`. `get_pse_data` now loads cache then downloads & appends data only newer than the cache. My problem is cache and every new phisix query have columns that match on `dt` and `close` only:

```python
data = get_stock_data(symbol="JFC", 
                    start_date="2018-01-01", 
                    end_date="2020-04-16", 
                    source="phisix",
                    format="dohlcv",
                   )
data.tail()
```

  dt | open | high | low | close | volume
-- | -- | -- | -- | -- | -- 
2020-04-08 | 111.0 | 120.0 | 108.1 | 120.0 | NaN
2020-04-13 | 121.0 | 136.1 | 121.0 | 135.0 | NaN
2020-04-14 | 139.9 | 148.0 | 139.8 | 146.5 | NaN
2020-04-15 | NaN | NaN | NaN | 148.6 | 3944620.0
2020-04-16 | NaN | NaN | NaN | 141.5 | 2318510.0

where 2020-04-14 is the last entry in cache.

Should we set `format="dc"` as new default, instead of the previous `format="dcv"`?
The advantage of having cache is enormous speedup (the query above took ~2 seconds) though we have to update the cache (takes ~3 minutes) say once a month so that new queries won't be longer than 30 iterations. Do we lose a lot if we ignore volume?

EDIT: I remember strategies require data format=DCV.  :(
Is volume here necessary or optional?

### CODE REVIEW

File: fastquant/fastquant.py [hunk 1]
```diff
@@ -142,10 +145,11 @@ def get_pse_data_old(
...
         Stock data (in OHLCV format) for the specified company and date range
     """
+    if stock_table_fp is None:
+        stock_table_fp = Path(DATA_PATH, "stock_table.csv")
 
-    if Path(DATA_PATH, stock_table_fp).exists():
-        print("Stock table exists!")
-        print("Reading {} ...".format(stock_table_fp))
+    if stock_table_fp.exists():
+        # print("Reading {} ...".format(stock_table_fp))
```
Comment (enzoampil) [inline fastquant/fastquant.py:152]: Can take out the comment

**Comment by enzoampil** [comment]
@jpdeleon Unfortunately, yes, volume is very important :( The biggest flaw of the old query function is it doesn't have volume. However, it shouldn't be an issue w/ fastquant, if you don't volume as long as you don't have an indicator that requires it.

The date format for strategies is "dcv" but this is configurable :) I'm okay with the above change, but can we set the default format as "dcv", with flexibility to add based on your specs?

**Comment by enzoampil** [comment]
@jpdeleon Can we apply `dcv` by default, and make sure "v" is the volume from phisix and note the old query function? If "v" is not used, should be fine.

**Comment by jpdeleon** [comment]
@enzoampil OK. I'll make a commit following your last suggestion.

**Comment by jpdeleon** [comment]
@enzoampil There are unrelated to cache but urgent bugfixes I had to fix here. See [examples/pse_data_cache.ipynb](https://github.com/enzoampil/fastquant/blob/cache/examples/stock_data_cache.ipynb) to see fix relevant to discussion above.

**Comment by enzoampil** [comment]
@jpdeleon 

Getting an error when running the ff:

```
from fastquant import *
data = get_stock_data('JFC', '2019-01-01', '2019-06-31', format='dcv')

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-17c0f9295091> in <module>()
----> 1 data = get_stock_data('JFC', '2019-01-01', '2019-06-31', format='dcv')

2 frames
/content/src/fastquant/fastquant/fastquant.py in get_stock_data(symbol, start_date, end_date, source, format)
    433     if source == "phisix":
    434         # The query is run on 'phisix', but if the symbol isn't found, the same query is run on 'yahoo'.
--> 435         df = get_pse_data(symbol, start_date, end_date)
    436         if df is None:
    437             df = get_yahoo_data(symbol, start_date, end_date)

/content/src/fastquant/fastquant/fastquant.py in get_pse_data(symbol, start_date, end_date, save, max_straight_nones)
    308     """
    309     start = datestring_to_datetime(start_date)
--> 310     end = datestring_to_datetime(end_date)
    311 
    312     cache = get_pse_data_cache(symbol=symbol)

/content/src/fastquant/fastquant/fastquant.py in datestring_to_datetime(date, sep)
    512     errmsg = "date format must be YYYY-MM-DD"
    513     assert len(ymd[0]) == 4, errmsg
--> 514     return datetime(*map(int, ymd))

ValueError: day is out of range for month

```

**Comment by jpdeleon** [comment]
@enzoampil June is until 30 :) Try
```
from fastquant import *
data = get_stock_data('JFC', '2019-01-01', '2019-06-30', format='dcv')
```

**Comment by enzoampil** [comment]
Sorry yeah, so querying with `dcv` format is still returning NaN's for volume.

```
data = get_stock_data('JFC', '2019-01-01', '2019-12-31', format='dcv')

#Missing columns filled w/ NaN: ['volume']


dt | close | volume
-- | -- | --
2019-01-02 | 293.0 | NaN
2019-01-03 | 292.0 | NaN
2019-01-04 | 309.0 | NaN
2019-01-07 | 323.0 | NaN
2019-01-08 | 321.0 | NaN
... | ... | ...
2019-12-19 | 216.0 | NaN
2019-12-20 | 213.0 | NaN
2019-12-23 | 220.0 | NaN
2019-12-26 | 216.2 | NaN
2019-12-27 | 216.0 | NaN



```

**Comment by jpdeleon** [comment]
@enzoampil I implemented a fix. See [examples/stock_data_cache.ipynb](https://github.com/enzoampil/fastquant/blob/cache/examples/stock_data_cache.ipynb) and please double check. Also phisix has a new endpoint: http://1.phisix-api.appspot.com/stocks (http://phisix-api2.appspot.com/stocks is now deprecated).

**Comment by enzoampil** [comment]
@jpdeleon Last thing, adding the `o` for "open" is returning NaN.

```
data = get_stock_data('JFC', '2019-01-01', '2019-12-31', format='dcov')
```

**Comment by jpdeleon** [comment]
@enzoampil strange... let me check

**Comment by jpdeleon** [comment]
@enzoampil `open` column is NaN because phisix is used whenever volume is requested. phisix does not have columns other than dcv. dohlc is possible however.

**Comment by enzoampil** [comment]
@jpdeleon Any chance we can merge `phisix` and `get_pse_data_old` results in case volume is asked for? :)

**Comment by jpdeleon** [comment]
@enzoampil I suggest to merge now because this branch already has several bugfixes

**Pull request merged**